interface/billing/sl_eob_invoice.php

```diff
@@ -108,51 +108,57 @@ function writeoff(code) {
-   continue;
-  }
-  var pfxlen = ename.indexOf('[pay]');
-  if (pfxlen < 0) continue;
-  var pfx = ename.substring(0, pfxlen);
-  var code = pfx.substring(pfx.indexOf('[')+1, pfxlen-1);
-  if (f[pfx+'[pay]'].value || f[pfx+'[adj]'].value) {
-   if (! f[pfx+'[date]'].value) {
-    alert('<?php xl('Date is missing for code ', 'e')?>' + code);
-    return false;
-   }
-  }
-  if (f[pfx+'[pay]'].value && isNaN(parseFloat(f[pfx+'[pay]'].value))) {
-   alert('<?php xl('Payment value for code ', 'e') ?>' + code + '<?php xl(' is not a number', 'e') ?>');
-   return false;
-  }
-  if (f[pfx+'[adj]'].value && isNaN(parseFloat(f[pfx+'[adj]'].value))) {
-   alert('<?php xl('Adjustment value for code ', 'e') ?>' + code + '<?php xl(' is not a number', 'e') ?>');
-   return false;
-  }
-  if (f[pfx+'[adj]'].value && ! f[pfx+'[reason]'].value) {
-   alert('<?php xl('Please select an adjustment reason for code ', 'e') ?>' + code);
-   return false;
-  }
-  // TBD: validate the date format
- }
- // Demand confirmation if deleting anything.
- if (delcount > 0) {
-  if (!confirm('<?php echo xl('Really delete'); ?> ' + delcount +
-   ' <?php echo xl('transactions'); ?>?' +
-   ' <?php echo xl('This action will be logged'); ?>!')
-  ) return false;
- }
- return true;
+    var delcount = 0; var allempty = true;
+    for (var i = 0; i < f.elements.length; ++i) {
+        var ename = f.elements[i].name;
+        // Count deletes.
+        if (ename.substring(0, 9) == 'form_del[') {
+            if (f.elements[i].checked) ++delcount;
+            continue;
+        }
+        var pfxlen = ename.indexOf('[pay]');
+        if (pfxlen < 0) continue;
+        var pfx = ename.substring(0, pfxlen);
+        var code = pfx.substring(pfx.indexOf('[') + 1, pfxlen - 1);
+        if (f[pfx + '[pay]'].value || f[pfx + '[adj]'].value) {
+            allempty = false;
+            if (!f[pfx + '[date]'].value) {
+                alert('<?php xl('Date is missing for code ', 'e')?>' + code);
+                return false;
+            }
+        }
+        if (f[pfx + '[pay]'].value && isNaN(parseFloat(f[pfx + '[pay]'].value))) {
+            alert('<?php xl('Payment value for code ', 'e') ?>' + code + '<?php xl(' is not a number', 'e') ?>');
+            return false;
+        }
+        if (f[pfx + '[adj]'].value && isNaN(parseFloat(f[pfx + '[adj]'].value))) {
+            alert('<?php xl('Adjustment value for code ', 'e') ?>' + code + '<?php xl(' is not a number', 'e') ?>');
+            return false;
+        }
+        if (f[pfx + '[adj]'].value && !f[pfx + '[reason]'].value) {
+            alert('<?php xl('Please select an adjustment reason for code ', 'e') ?>' + code);
+            return false;
+        }
+        // TBD: validate the date format
+    }
+    // Check if save is clicked with nothing to post.
+    if (allempty && delcount === 0) {
+        alert('<?php xl('Nothing to Post! Please review entries or use Cancel to exit transaction', 'e')?>');
```

Comment:
use echo xls()
(and drop the e parameter)

Reply:
Trying to keep the code consistent here as all the other alerts use the older translation in this fashion. Not really sure an echo is needed in this case because this acts more like a constant where php returns the string to the quote. If you feel really strongly then I will change but this tested well and prefer to leave as is.

Reply:
Minor issue, so can disregard this here.
For new code, generally rec to use the newer security methods and not use the e parameter, which is deprecated; this was partly done because it bypasses html escaping.